js/utils/AStar.js

```diff
@@ -0,0 +1,262 @@
+class Node {
+	constructor(tile, parentNode = null) {
+		// keep the tile represented by this node
+		this.tile = tile;
+		// unique id
+		this.id = tile.y * Node.MAP_SIZES[Node.CURR_MAP_NAME].WIDTH + tile.x;
+
+		this.parentNode = parentNode;
+
+		// f = h + g in AStar algorithm where h is the heuristic function value
+		// in our case h = distance from this tile to the endTile
```

Comment:
why did u choose this heuristic?

Reply:
It's actually f = h = distanceFunction. The distance function chosen is the euclidean distance function which behaves pretty good for 8 directional movement. It also decreases faster than the manhattan distance (I tested both).